Assets/Scripts/Actors/ActorComponents/ActorPhysics.cs

```diff
@@ -12,28 +12,32 @@ public enum ActorStates
 
 public class ActorPhysics : ActorComponent
 {
+	private new Transform transform;
```

Comment:
It's just weird to me that it depends on Awake for transform to work. If anything else tries to refer to this before awake it will get a null ref. I'd prefer something separate like myTransform so the getComponent functionality is still there as well.